monai/apps/generation/maisi/networks/autoencoderkl_maisi.py

```diff
@@ -0,0 +1,929 @@
+            print("padding:", padding)
+
+        overlaps = [0] + [padding] * (num_splits - 1)
+        last_padding = x.size(self.dim_split + 2) % split_size
+
+        if self.dim_split == 0:
+            splits = [
+                x[
+                    :,
+                    :,
+                    i * split_size
+                    - overlaps[i] : (i + 1) * split_size
+                    + (padding if i != num_splits - 1 else last_padding),
+                    :,
+                    :,
+                ]
+                for i in range(num_splits)
+            ]
+        elif self.dim_split == 1:
+            splits = [
+                x[
+                    :,
+                    :,
+                    :,
+                    i * split_size
+                    - overlaps[i] : (i + 1) * split_size
+                    + (padding if i != num_splits - 1 else last_padding),
+                    :,
+                ]
+                for i in range(num_splits)
+            ]
+        elif self.dim_split == 2:
+            splits = [
+                x[
+                    :,
+                    :,
+                    :,
+                    :,
+                    i * split_size
+                    - overlaps[i] : (i + 1) * split_size
+                    + (padding if i != num_splits - 1 else last_padding),
+                ]
+                for i in range(num_splits)
+            ]
+
+        if self.debug:
+            for j in range(len(splits)):
+                print(f"splits {j + 1}/{len(splits)}:", splits[j].size())
+
+        del x
+        _empty_cuda_cache()
+
+        outputs = [self.conv(split) for split in splits]
+
+        if self.debug:
+            for j in range(len(outputs)):
+                print(f"outputs before {j + 1}/{len(outputs)}:", outputs[j].size())
+
+        split_size_out = split_size
+        padding_s = padding
+        non_dim_split = self.dim_split + 1 if self.dim_split < 2 else 0
+        if outputs[0].size(non_dim_split + 2) // splits[0].size(non_dim_split + 2) == 2:
+            split_size_out *= 2
+            padding_s *= 2
+        elif splits[0].size(non_dim_split + 2) // outputs[0].size(non_dim_split + 2) == 2:
+            split_size_out //= 2
+            padding_s //= 2
+
+        if self.dim_split == 0:
+            outputs[0] = outputs[0][:, :, :split_size_out, :, :]
+            for i in range(1, num_splits):
+                outputs[i] = outputs[i][:, :, padding_s : padding_s + split_size_out, :, :]
+        elif self.dim_split == 1:
+            outputs[0] = outputs[0][:, :, :, :split_size_out, :]
+            for i in range(1, num_splits):
+                outputs[i] = outputs[i][:, :, :, padding_s : padding_s + split_size_out, :]
+        elif self.dim_split == 2:
+            outputs[0] = outputs[0][:, :, :, :, :split_size_out]
+            for i in range(1, num_splits):
+                outputs[i] = outputs[i][:, :, :, :, padding_s : padding_s + split_size_out]
```

Comment:
```suggestion
        slices = [slice(None)] * 5
        for i in range(1, num_splits):
            slices[self.dim_split+2]=slice(None,split_size_out) if i == 0 else slice(padding_s, padding_s + split_size_out)
            outputs[i] = outputs[i][tuple(slices)]
```
I think this is equivalent...

Reply:
Thanks! @ericspod I refined the script based on your suggestion.

Reply:
Hi @ericspod , I have updated the script. Shall we resolve the thread? Thanks.